api/controller_test.go

```diff
@@ -1201,6 +1201,87 @@ func TestController_ObjectsUploadObjectHandler(t *testing.T) {
 	})
 }
 
+func TestController_ObjectsStageObjectHandler(t *testing.T) {
+	clt, deps := setupClient(t, "s3")
+
+	// create user
+	creds := createDefaultAdminUser(t, clt)
+	bauth := httptransport.BasicAuth(creds.AccessKeyID, creds.AccessSecretKey)
+
+	ctx := context.Background()
+	_, err := deps.cataloger.CreateRepository(ctx, "repo1", "s3://bucket/prefix", "master")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Run("stage object", func(t *testing.T) {
+		resp, err := clt.Objects.StageObject(objects.NewStageObjectParams().
+			WithRepository("repo1").
+			WithBranch("master").
+			WithPath("foo/bar").
+			WithStats(&models.ObjectStats{
+				Checksum:        "afb0689fe58b82c5f762991453edbbec",
+				Path:            "foo/bar",
+				PhysicalAddress: "s3://another-bucket/some/location",
+				SizeBytes:       swag.Int64(38),
+			}), bauth)
+
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if swag.Int64Value(resp.Payload.SizeBytes) != 38 {
+			t.Fatalf("expected 38 bytes to be written, got back %d", resp.Payload.SizeBytes)
+		}
+
+		// get back info
+		got, err := clt.Objects.StatObject(objects.NewStatObjectParams().
+			WithRepository("repo1").
+			WithRef("master").
+			WithPath("foo/bar"), bauth)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if got.Payload.PhysicalAddress != "s3://another-bucket/some/location" {
+			t.Fatalf("unexpected physical address: %s", got.Payload.PhysicalAddress)
+		}
+	})
+
+	t.Run("upload object missing branch", func(t *testing.T) {
```

Comment:
Can we also test for no such repo?  We've had bugs for being precise on that error, recently #1476.